Unable to import okta_email_template_settings

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform 1.6.0
Provider 4.13.1

### Affected Resource(s)

- `okta_email_template_settings`

### Terraform Configuration Files

```hcl
resource "okta_email_template_settings" "example" {
  brand_id      = "<brand id>"
  template_name = "ForgotPassword"
  recipients    = "ADMINS_ONLY"
}
```

### Debug Output

```
│ Error: failed to read email template settings
│ 
│ 405 Method Not Allowed
```

Relevant debug except showing the endpoint the provider attempts to call (the correct endpoint is `/api/v1/brands/{{brandID}}/templates/email/{{templateType}}/settings`)

```
2025-01-31T14:18:11.728-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: 2025/01/31 02:18:11 [DEBUG] performing request: method=GET url={{tenant URL}}/api/v1/brands//templates/email//settings
2025-01-31T14:18:11.728-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: 2025/01/31 14:18:11 [DEBUG] Okta API Request Details:
2025-01-31T14:18:11.728-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: ---[ REQUEST ]---------------------------------------
2025-01-31T14:18:11.728-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: GET /api/v1/brands//templates/email//settings HTTP/1.1

2025-01-31T14:18:11.836-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: 2025/01/31 14:18:11 [DEBUG] Okta API Response Details:
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: ---[ RESPONSE ]--------------------------------------
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: HTTP/2.0 405 Method Not Allowed
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: Connection: close
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: Content-Type: application/json
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: Date: Fri, 31 Jan 2025 19:18:11 GMT
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: Server: nginx
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: Strict-Transport-Security: max-age=315360000; includeSubDomains
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: X-Content-Type-Options: nosniff
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: {
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1:  "errorCode": "E0000022",
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1:  "errorSummary": "The endpoint does not support the provided HTTP method",
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1:  "errorLink": "E0000022",
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1:  "errorId": "oaeiSS5fQJWRr6Bc9Cx7mjkWw",
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1:  "errorCauses": []
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: }
2025-01-31T14:18:11.837-0500 [DEBUG] provider.terraform-provider-okta_v4.13.1: -----------------------------------------------------
2025-01-31T14:18:11.837-0500 [ERROR] provider.terraform-provider-okta_v4.13.1: Response contains error diagnostic: diagnostic_summary="failed to read email template settings" tf_proto_version=5.4 @caller=github.com/hashicorp/terraform-plugin-go@v0.22.2/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto tf_rpc=ReadResource diagnostic_detail="405 Method Not Allowed" diagnostic_severity=ERROR tf_req_id=1a75007e-bd24-265a-cac1-92f493eab083 tf_resource_type=okta_email_template_settings tf_provider_addr=okta/okta timestamp=2025-01-31T14:18:11.836-0500
2025-01-31T14:18:11.838-0500 [ERROR] vertex "import okta_email_template_settings.example result" error: failed to read email template settings
```

### Panic Output

N/A

### Expected Behavior

The resource is imported and managed by Terraform

#### Can this be done in the Admin UI?

N/A

#### Can this be done in the actual API call?

N/A

### Actual Behavior

The import fails

### Steps to Reproduce

- Define a `okta_email_template_settings` resource
- Attempt to import it using the ID `<brand ID>/<template type>` per the [provider doc](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/email_template_settings#import)
- Observe that the import fails

### Important Factoids

Looking at the [import method](https://github.com/okta/terraform-provider-okta/blob/master/okta/resource_okta_email_template_settings.go#L151) and [framework docs](https://developer.hashicorp.com/terraform/plugin/framework/resources/import#multiple-attributes) for the `ImportStatePassthroughID` method it uses, this resource isn't compatible with it, because the resource's read method relies on multiple inputs (brand ID & template type).

Looks like a simple fix (define a custom import method that extracts the two inputs in the provided composite ID), it's not a major blocker for this project, so I'll circle back and put in a PR if I can find the time.

### References

--- a/okta/resource_okta_email_template_settings.go
+++ b/okta/resource_okta_email_template_settings.go
@@ -3,8 +3,8 @@ package okta
 import (
 	"context"
 	"fmt"
+	"strings"
 
-	"github.com/hashicorp/terraform-plugin-framework/path"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
@@ -149,7 +149,17 @@ func (r *emailTemplateSettingsResource) Update(ctx context.Context, req resource
 }
 
 func (r *emailTemplateSettingsResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+	idParts := strings.Split(req.ID, "/")
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		resp.Diagnostics.AddError("Invalid Import Identifier", "Expected import identifier with format <brand_id>/<template_name>")
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &emailTemplateSettingsResourceModel{
+		ID:           types.StringValue(req.ID),
+		BrandID:      types.StringValue(idParts[0]),
+		TemplateName: types.StringValue(idParts[1]),
+	})...)
 }
 
 func (r *emailTemplateSettingsResource) put(ctx context.Context, plan emailTemplateSettingsResourceModel) error {

--- a/okta/resource_okta_email_template_settings_test.go
+++ b/okta/resource_okta_email_template_settings_test.go
@@ -1,10 +1,12 @@
 package okta
 
 import (
+	"errors"
 	"fmt"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
 func TestAccResourceOktaEmailTemplateSettings_crud(t *testing.T) {
@@ -32,6 +34,26 @@ func TestAccResourceOktaEmailTemplateSettings_crud(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "recipients", "ADMINS_ONLY"),
 					resource.TestCheckResourceAttr(resourceName, "template_name", "UserActivation"),
 				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					rs, ok := s.RootModule().Resources[resourceName]
+					if !ok {
+						return "", fmt.Errorf("failed to find %s", resourceName)
+					}
+					brandID := rs.Primary.Attributes["brand_id"]
+					templateName := rs.Primary.Attributes["template_name"]
+					return fmt.Sprintf("%s/%s", brandID, templateName), nil
+				},
+				ImportStateCheck: func(s []*terraform.InstanceState) error {
+					if len(s) != 1 {
+						return errors.New("failed to import schema into state")
+					}
+					return nil
+				},
 			},
 		},
 	})

